ui/apps/platform/src/Components/CodeViewer.tsx

```diff
@@ -0,0 +1,125 @@
+import React, {
+    createContext,
+    CSSProperties,
+    Dispatch,
+    ReactNode,
+    SetStateAction,
+    useContext,
+    useState,
+} from 'react';
+import { CodeBlockAction, ClipboardCopyButton, Button, CodeBlock } from '@patternfly/react-core';
+import { MoonIcon, SunIcon } from '@patternfly/react-icons';
+
+import { PrismLight as SyntaxHighlighter } from 'react-syntax-highlighter';
+import lightTheme from 'react-syntax-highlighter/dist/esm/styles/prism/one-light';
+import darkTheme from 'react-syntax-highlighter/dist/esm/styles/prism/one-dark';
+import yaml from 'react-syntax-highlighter/dist/esm/languages/prism/yaml';
+
+import useClipboardCopy from 'hooks/useClipboardCopy';
+
+SyntaxHighlighter.registerLanguage('yaml', yaml);
+
+const CodeViewerThemeContext = createContext<
+    ['light' | 'dark', Dispatch<SetStateAction<'light' | 'dark'>>] | undefined
+>(undefined);
+
+export const CodeViewerThemeProvider = ({ children }) => {
+    const [state, setState] = useState<'light' | 'dark'>('light');
```

Comment:
To keep my mind alert on Friday afternoon, what (if any) changes do you foresee here if entire UI supports light/dark mode?

Reply:
I'd say minimum we would need to have the default theme match whatever the user's current theme is.

Whether or not a global light/dark theme would remove the need to toggle to code viewer independently is worthwhile to discuss when the time comes.